worlds/tracker/docs/apworld-integration.md

```diff
@@ -0,0 +1,98 @@
+
+UT does not have access to the seed used during the original generation. This means that any randomization that is not a direct result of YAML options or AP items will not be the same when UT runs and must be provided in slot data. This can include entrance rando (GER or otherwise), level order, starting location, or any similar options that don't use items directly.
+
+Best practice is to store these random results on your world instance and pass them into the result of `fill_slot_data`.
+
+```python
+    def fill_slot_data(self) -> dict[str, Any]:
+        return {
+            "starting_location": self.starting_location,
+            "entrances": self.randomized_entrances, # eg the result of calling randomize_entrances(...).pairings
+            # Add any as needed
+        }
+```
+
+## Loading provided information
+
+You can access the slot data from the original generation by defining an `interpret_slot_data` function in your world. UT will call this function once connected to the multiworld with the slot data it recieved for that slot. Using this data, you can adjust any world state that affects your rules, such as world instance attributes or entrance connections.
+
+```python
+    def interpret_slot_data(self, slot_data: dict[str, Any]) -> None:
+        if "starting_location" in slot_data:
+            self.origin_region_name = slot_data["starting_location"]
+
+        if "entrances" in slot_data:
+            # Update entrance connections for ER
+            entrances = {
+                entrance.name: entrance
+                for region in self.get_regions()
+                for entrance in region.entrances
+            }
+            for source_exit, target_entrance in slot_data["entrances"]:
+                entrances[source_exit].connected_region = entrances[target_entrance].parent_region
+```
+
+You may also want to adjust your world's behaviour in other functions to reduce the amount of work you need to do in `interpret_slot_data`. To do so, you can check the `generation_is_fake` attribute on the `multiworld` object, which will be `True` if your world is generating inside UT. You can use this to skip doing things that will be handled by `interpret_slot_data` later, or to run code that would normally be skipped based on options. For example, you can create every possible location and let UT filter them out to just what exists from the server so you don't need to worry about creating them later.
+
+```python
+    def create_regions(self) -> None:
+        is_ut = getattr(self.multiworld, "generation_is_fake", False)
+        if self.randomly_rolled_property == 1 or is_ut:
+            # This location will always be created by UT regardless of what UT happens to roll
+            self.create_location(...)
+```
+
+## Generating without a YAML
+
+You can also make it so a YAML is not required to generate for your world. To do so, you must first store all options that affect generation in your slot data.
```

Comment:
```suggestion
You can also make it so a YAML is not required to generate for your world. To do so, you must first store all options that affect generation in your slot data. Take care not to include options that don't affect generation and aren't useful for the game client.
```
I know the first sentence already implies this, but I feel like it should be emphasized a bit.